wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -136,16 +175,114 @@ class PresentationActivity : FragmentActivity() {
         }
     }
 
-    override fun onDestroy() {
-        Logger.i(TAG, "onDestroy")
-        disconnect()
-        super.onDestroy()
+    /**
+     * Enum class defining the types of result statuses that a Presentment can yield to.
+     *
+     * @param messageResourceId the message to show for a status
+     * @param imageResourceId the image to show above the message
+     */
+    enum class ResultStatus(val messageResourceId: Int, val imageResourceId: Int) {
+        // All prompts were successful
+        SUCCESS(R.string.presentation_result_success_message, R.drawable.check_48x48),
+
+        // Something prevented success
+        ERROR(R.string.presentation_result_error_message, R.drawable.error_48x48),
+
+        // For init of Flow
+        INIT(0, 0)
+    }
+
+    private val _resultStatus = MutableStateFlow(ResultStatus.INIT)
+    private val resultStatus: StateFlow<ResultStatus> = _resultStatus
+
+    /**
+     * Composable view that is shown at the end of the Presentation result to briefly update the user
+     * on the result of the Presentation Prompt - success, neutral, or error.
+     * @param resultStatus the [ResultStatus] object containing the result status data to show.
+     */
+    @Composable
+    fun ResultStatus(resultStatus: ResultStatus) {
+        val activity = LocalContext.current as? Activity
+        val coroutineScope = rememberCoroutineScope() // Get the coroutine scope
+
+        var visible by remember { mutableStateOf(true) }
+        if (resultStatus != ResultStatus.INIT) {
```

Comment:
Cleaner to just invert the logic and return. That way we don't have to indent the stuff below.